tests/integration/sql_session_manager.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import asdict, dataclass
+from functools import cached_property
+from typing import Any, Type
+
+import sqlalchemy as sa
+from sqlalchemy.pool import Pool, QueuePool
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ConnectionDetails:
+    connection_string: str
+
+    @cached_property
+    def dialect(self) -> str:
+        return sa.engine.make_url(self.connection_string).get_backend_name()
+
+
+@dataclass(frozen=True)
+class PoolConfig:
```

Comment:
It would be nice to add a docstring indicating that these parameters (or others for the future) can be found [here](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params).